src/DTO/UserEditDTO.php

```diff
@@ -0,0 +1,92 @@
+<?php
+
+namespace App\DTO;
+
+use DateTime;
+use Doctrine\Common\Collections\Collection;
+use JMS\Serializer\Annotation as Serializer;
+use Symfony\Component\Validator\Constraints as Assert;
+
+class UserEditDTO
```

Comment:
Don't duplicate the DTO. Use one DTO with serializer groups/context. Asserts also have groups when to be applied and when not.